executor/trace.go

```diff
@@ -93,44 +96,74 @@ func (e *TraceExec) Next(ctx context.Context, req *chunk.Chunk) error {
 }
 
 func (e *TraceExec) nextOptimizerPlanTrace(ctx context.Context, se sessionctx.Context, req *chunk.Chunk) error {
-	zf, fileName, err := generateOptimizerTraceFile()
-	if err != nil {
-		return err
-	}
-	zw := zip.NewWriter(zf)
-	defer func() {
-		err := zw.Close()
-		if err != nil {
-			logutil.BgLogger().Warn("Closing zip writer failed", zap.Error(err))
-		}
-		err = zf.Close()
-		if err != nil {
-			logutil.BgLogger().Warn("Closing zip file failed", zap.Error(err))
-		}
-	}()
-	traceZW, err := zw.Create("trace.json")
-	if err != nil {
-		return errors.AddStack(err)
-	}
 	stmtCtx := se.GetSessionVars().StmtCtx
-	origin := stmtCtx.EnableOptimizeTrace
-	stmtCtx.EnableOptimizeTrace = true
+
```

Comment:
I‘m afraid this function might become too complex to maintain, would you separate it by different opttimize trace target?

Reply:
Updated.